p2p/pex/pex_reactor.go

```diff
@@ -372,6 +364,37 @@ func (r *PEXReactor) ensurePeers() {
 	}
 }
 
+func (r *PEXReactor) dialPeer(addr *p2p.NetAddress) {
+	// 1s == (1e9 ns) == (1 Billion ns)
+	billionNs := float64(time.Second.Nanoseconds())
+
+	// exponential backoff if it's not our first attempt to dial given address
+	var attempts int
+	if lAttempts, attempted := r.attemptsToDial.Load(addr.DialString()); attempted {
+		attempts = lAttempts.(int)
```

Comment:
Can we test this behaviour? I think this branch isn't covered with our current tests. It might also surface that we want to make the jitter duration configurable in some form.

Reply:
How about now? https://github.com/tendermint/tendermint/pull/1292/commits/f85c8896d9877f034e6c40d11887637150940841

Reply:
Beautiful and simple!